### PR TITLE
Split content of interface.ts

### PR DESCRIPTION
**Pull request by SteffenKn**
## What did you change?

- All interfaces are now located in separate files.

## How can others test the changes?

- Look through the source folder, you'll immediately see the changes.
- Build a solution that uses the metadata_contracts project. The build should run exactly as before, since all changes made were exclusively cosmetic and non-functional. 

## PR-Checklist

Please check the boxes in this list after submitting your PR:

- [x] You can merge this PR **right now** (if not, please prefix the title with "WIP: ")
- [x] I've tested **all** changes included in this PR.
- [x] I've also reviewed this PR myself before submitting (e.g. for scrambled letters, typos, etc.)
- [x] I've merged the `develop` branch into my branch before finishing this PR.
- [x] I've **not added any other changes** than the ones described above.
- [x] I've mentioned all **PRs, which relate to this one**

### CODE REVIEW

File: src/inamespace_decorators.ts [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+import {IRequiresClaimDecorator} from './irequires_claim_decorator';
+import {IPromoteDecorator} from './ipromote_decorator';
+
+
+export interface INamespaceDecorators {
+  schemaAttribute(schema: ISchemaAttribute): ISchemaAttributeDecorator;
+  schemaClass(schema: ISchemaClass): ISchemaClassDecorator;
+  requiresClaim(claim: IRequiredClaim): IRequiresClaimDecorator;
+  promote(roles: IPromotionRoles): IPromoteDecorator;
+}
```
Comment (ElRaptorus) [inline src/inamespace_decorators.ts]: Final Newline missing

File: src/index.ts [hunk 2]
```diff
@@ -1 +1,7 @@
-export * from './interfaces';
\ No newline at end of file
+export * from './imetadata_provider';
+export * from './inamespace_decorators';
+export * from './ipromote_decorator';
+export * from './irequires_claim_decorator';
+export * from './ischema_attribute_decorator';
+export * from './ischema_class_decorator';
+export * from './metadata_type';
```
Comment (ElRaptorus) [inline src/index.ts]: Final Newline missing

File: src/ipromote_decorator.ts [hunk 3]
```diff
@@ -0,0 +1,3 @@
+export interface IPromoteDecorator {
+  (...args: Array<any>): typeof args[0] | void | PropertyDescriptor;
+}
```
Comment (ElRaptorus) [inline src/ipromote_decorator.ts]: Final Newline missing

File: src/irequires_claim_decorator.ts [hunk 4]
```diff
@@ -0,0 +1,3 @@
+export interface IRequiresClaimDecorator {
+  (...args: Array<any>): typeof args[0] | void | PropertyDescriptor;
+}
```
Comment (ElRaptorus) [inline src/irequires_claim_decorator.ts]: Final Newline missing

File: src/ischema_attribute_decorator.ts [hunk 5]
```diff
@@ -0,0 +1,3 @@
+export interface ISchemaAttributeDecorator {
+  (...args: Array<any>): void | PropertyDescriptor;
+}
```
Comment (ElRaptorus) [inline src/ischema_attribute_decorator.ts]: Final Newline missing

File: src/ischema_class_decorator.ts [hunk 6]
```diff
@@ -0,0 +1,3 @@
+export interface ISchemaClassDecorator {
+  (...args: Array<any>): typeof args[0];
+}
```
Comment (ElRaptorus) [inline src/ischema_class_decorator.ts]: Final Newline missing

File: src/metadata_type.ts [hunk 7]
```diff
@@ -0,0 +1,6 @@
+export enum MetadataType {
+  RequiredClaim,
+  SchemaAttribute,
+  SchemaClass,
+  Promotion,
+}
```
Comment (ElRaptorus) [inline src/metadata_type.ts]: Final Newline missing

**Comment by Paulomart** [comment]
@SteffenKn Your git email address is `Steffen@Steffens-MacBook-Pro.fritz.box`. This doesnt look right. You can set your email address globally by using `git config --global user.email <email>`

**Pull request merged**